clientd3d/color.c

```diff
@@ -451,21 +451,26 @@ COLORREF GetPlayerNameColor(int flags,const char*name)
 {
 	if (GetDrawingEffect(flags) == OF_BLACK)
 		return NAME_COLOR_BLACK_FG;
-
-	switch (GetPlayerFlags(flags))
-	{
+	
+	/* If morphed, temporarily add OF_PLAYER for name coloring */
+    int color_flags = flags;
+    if (flags & OF_MORPHED)
+        color_flags |= OF_PLAYER;
```

Comment:
What would be the implication of *always* turning on this bit when getting the player color?  I'm not super excited about adding OF_MORPHED in the protocol, because it means that cheaters could use it to detect someone who is morphed.  Is that a problem?